tools/syz-declextract/syz-declextract.cpp

```diff
@@ -142,41 +177,117 @@ const std::string makeConst(const std::string &type, const std::string &val = "0
   if (type.empty()) {
     return "const[" + val + "]";
   }
-  if (type == "void") {
-    return "const[" + val + ", intptr]";
-  }
   return "const[" + val + ", " + type + "]";
 }
 
-const std::string getSyzType(const std::string &ctype, const bool isSyscallParam = false) {
+enum IntType {
+  INVALID_INT = 0,
+  INT_8 = 1,
+  INT_16 = 2,
+  INT_32 = 4,
+  INT_64 = 8,
+  INT_PTR,
+};
+
+IntType getIntType(const std::string &ctype, const bool isSyscallParam) {
+  // TODO: Handle arm32 passing 64bit arguments
   if (!isSyscallParam && (contains(ctype, "long long") || contains(ctype, "64"))) {
-    return contains(ctype, "be") ? "int64be" : "int64";
+    return INT_64;
   }
   if (contains(ctype, "16") || contains(ctype, "short")) {
-    return contains(ctype, "be") ? "int16be" : "int16";
+    return INT_16;
   }
-  if (contains(ctype, "8")) {
-    return contains(ctype, "be") ? "int8be" : "int8";
+  if (contains(ctype, "8") || contains(ctype, "char")) {
+    return INT_8;
   }
   if (contains(ctype, "32") || contains(ctype, "int")) {
-    return contains(ctype, "be") ? "int32be" : "int32";
+    return INT_32;
+  }
+  if (contains(ctype, "long")) {
+    return INT_PTR;
+  }
+  fprintf(stderr, "Unhandled int length for type: %s\n", ctype.c_str());
+  exit(1);
+}
+
+const std::string intNCommonSubtype(const std::string &name, const IntType len) {
+  const auto &byteLen = std::to_string(len * 8);
+  if (endsWith(name, "enabled") || endsWith(name, "enable")) {
+    return "bool" + byteLen;
+  }
+  return "int" + byteLen;
+}
+
+const std::string intNSubtype(const std::string &lowerCaseName, const IntType len) {
+  switch (len) {
+  case INT_8:
+    return int8Subtype(lowerCaseName);
+  case INT_16:
+    return int16Subtype(lowerCaseName);
+  case INT_32:
+    return int32Subtype(lowerCaseName);
+  case INT_64:
+    return int64Subtype(lowerCaseName);
+  default:
+    fprintf(stderr, "Invalid int type\n");
+    exit(1);
+  }
+}
+
+bool isIntN(const std::string &syztype) {
+  return !syztype.compare(0, 3, "int") && std::all_of(syztype.begin() + 3, syztype.end(), ::isDigit);
+}
+
+const std::string intSubtype(std::string name, const IntType len) {
+  if (len == INVALID_INT) {
+    fprintf(stderr, "Invalid int type\n");
+    exit(1);
+  } else if (len == INT_PTR) {
+    return "intptr";
+  }
+
+  std::transform(name.begin(), name.end(), name.begin(), ::tolower);
```

Comment:
This is still placed in 3 semi-random places. It's hard to understand looking at some function, if the name is already lowered there or not. It may be lowered on some call chains, but not on others. It's also easy add bug assuming the name is lowered, when it's not. It's hard to understand if I need to add lowering in more places, if I am calling some function from a new location.

It will be better to do it in the beginning of nlaToSyz and getSyzType.
Then the invariant is simple: the name is already lowered in all type analysis code, no new lowered needs to be added. Hard to introduce bugs.